homeassistant/components/script.py

```diff
@@ -42,6 +44,62 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+_ALIAS_VALIDATOR = vol.Schema(cv.string)
+
+
+def _alias_stripper(validator):
+    """Strip alias from object for validation."""
+    def validate(value):
+        """Validate without alias value."""
+        value = value.copy()
+        alias = value.pop(CONF_ALIAS, None)
+
+        if alias is not None:
+            alias = _ALIAS_VALIDATOR(alias)
+
+        value = validator(value)
+
+        if alias is not None:
+            value[CONF_ALIAS] = alias
+
+        return value
+
+    return validate
+
+
+_DELAY_SCHEMA = {
+    vol.Required(CONF_DELAY): vol.All({
+        CONF_ALIAS: str,
+        'days': vol.All(vol.Coerce(int), vol.Range(min=0)),
+        'seconds': vol.All(vol.Coerce(int), vol.Range(min=0)),
+        'microseconds': vol.All(vol.Coerce(int), vol.Range(min=0)),
+        'milliseconds': vol.All(vol.Coerce(int), vol.Range(min=0)),
+        'minutes': vol.All(vol.Coerce(int), vol.Range(min=0)),
+        'hours': vol.All(vol.Coerce(int), vol.Range(min=0)),
+        'weeks': vol.All(vol.Coerce(int), vol.Range(min=0)),
+    }, cv.has_at_least_one_key([
+        'days', 'seconds', 'microseconds', 'milliseconds', 'minutes', 'hours',
+        'weeks']))
+}
+
+_EVENT_SCHEMA = cv.EVENT_SCHEMA.extend({
+    CONF_ALIAS: cv.string,
```

Comment:
In the `_DELAY_SCHEMA` and the `_SCRIPT_ENTRY_SCHEMA` the aliases are defined as simple 'str' type, but here it it coerced to a string. I would expect them to be treated the same everywhere.